bundle/regal/rules/idiomatic/use_in_operator.rego

```diff
@@ -9,35 +9,44 @@ import data.regal.result
 report contains violation if {
 	some terms in eq_exprs_terms
 
-	terms[1].type in {"array", "boolean", "object", "null", "number", "set", "string", "var"}
-	terms[2].type == "ref"
+	nl_terms := non_loop_term(terms)
+	count(nl_terms) == 1
 
-	last := regal.last(terms[2].value)
+	nlt := nl_terms[0]
+	static_term(nlt.term)
 
-	last.type == "var"
-	startswith(last.value, "$")
-
-	violation := result.fail(rego.metadata.chain(), result.location(terms[2].value[0]))
+	# Use the non-loop term positon to determine the
+	# location of the loop term (3 is the count of terms)
+	violation := result.fail(rego.metadata.chain(), result.location(terms[3 - nlt.pos]))
 }
 
-report contains violation if {
-	some terms in eq_exprs_terms
+eq_exprs_terms contains terms if {
+	terms := input.rules[_].body[_].terms
 
-	terms[1].type == "ref"
-	terms[2].type in {"array", "boolean", "object", "null", "number", "set", "string", "var"}
+	terms[0].type == "ref"
+	terms[0].value[0].type == "var"
+	terms[0].value[0].value == "equal"
+}
 
-	last := regal.last(terms[1].value)
+non_loop_term(terms) := [{"pos": i + 1, "term": term} |
+	some i, term in array.slice(terms, 1, 3)
+	not loop_term(term)
+]
 
+loop_term(term) if {
+	term.type == "ref"
+	term.value[0].type == "var"
+	last := regal.last(term.value)
 	last.type == "var"
 	startswith(last.value, "$")
-
-	violation := result.fail(rego.metadata.chain(), result.location(terms[1].value[0]))
 }
 
-eq_exprs_terms contains terms if {
-	terms := input.rules[_].body[_].terms
+static_term(term) if term.type in {"array", "boolean", "object", "null", "number", "set", "string", "var"}
 
-	terms[0].type == "ref"
-	terms[0].value[0].type == "var"
-	terms[0].value[0].value == "equal"
+static_term(term) if {
+	term.type == "ref"
+
+	every t in array.slice(term.value, 1, count(term.value)) {
+		t.type == "string"
```

Comment:
💭 What about non-string scalars?
```rego
p[x].foo = "bar" if x := true
p[x].foo = "bar" if x := 1
p[x].foo = "bar" if x := set()
```
⏬ 
```json
{
    "p": {
        "1": {
            "foo": "bar"
        },
        "[]": {
            "foo": "bar"
        },
        "true": {
            "foo": "bar"
        }
    }
}
```
not the most common thing, but still "static".

Reply:
That's true! So rather than checking for "string", we could do:
```rego
every t in array.slice(ref.value, 1, count(ref.value)) {
    t.type != "var"
}
```
?

Reply:
:tada: every-not-p is not-some-p, so I guess there are more potential places 😉